File: SDE_sheet/Arrays/04_kadane_algorithm/solution.ts
function maxSubArray(nums: number[]): number {
    let n: number = nums.length
    let max: number = Number.MIN_SAFE_INTEGER
    let curr: number = 0
    for(const num of nums){
        curr = Math.max(curr + num, num)
        max = Math.max(max, curr)
    }
    return max
};

File: SDE_sheet/Arrays/04_kadane_algorithm/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length, max = Integer.MIN_VALUE, curr = 0;
        for (int num : nums) {
            curr = Math.max(curr + num, num);
            max = Math.max(max, curr);
        }
        return max;
    }
}
